addons/binding/org.openhab.binding.tesla/src/main/java/org/openhab/binding/tesla/handler/TeslaHandler.java

```diff
@@ -0,0 +1,948 @@
+					.queryParam(
+							"values",
+							StringUtils.join(EventKeys.values(), ',', 1,
+									EventKeys.values().length));
+			eventResponse = eventTarget.request(MediaType.TEXT_PLAIN_TYPE)
+					.get();
+			if (eventResponse.getStatus() == 200) {
+				InputStream dummy = (InputStream) eventResponse.getEntity();
+				// eventInputStreamReader = new
+				// InputStreamReader(eventResponse.readEntity(InputStream.class));
+				eventInputStreamReader = new InputStreamReader(dummy);
+				eventBufferedReader = new BufferedReader(eventInputStreamReader);
+				return true;
+			} else {
+				return false;
+			}
+		}
+
+		@Override
+		public void run() {
+			try {
+				if (isAwake()) {
+					if (eventBufferedReader == null
+							|| (!isInMotion() && (System.currentTimeMillis()
+									- lastEventSystemTime > MINIMUM_EVENT_INTERVAL))) {
+						try {
+							if (!establishEventStream()) {
+								vehicle = queryVehicle();
+								eventBufferedReader = null;
+							}
+						} catch (Exception e) {
+							logger.error(
+									"An exception occurred while establishing the event stream for the vehicle: '{}'",
+									e.getMessage());
+							vehicle = queryVehicle();
+							eventBufferedReader = null;
+						}
+					}
+
+					try {
+						if (eventBufferedReader != null) {
+							String line = null;
+							try {
+								line = eventBufferedReader.readLine();
+							} catch (Exception e) {
+								// we just move on. If we are here, then is most
+								// probably due to Premature EOF exceptions
+							}
+							if (line != null) {
+								lastEventSystemTime = System
+										.currentTimeMillis();
+								logger.debug("Received an event: '{}'", line);
+								String vals[] = line.split(",");
+								if (!vals[0].equals(lastEventTimeStamp)) {
+									lastEventTimeStamp = vals[0];
+									for (int i = 0; i < EventKeys.values().length; i++) {
+										try {
+											TeslaChannelSelector selector = TeslaChannelSelector
+													.getValueSelectorFromRESTID((EventKeys
+															.values()[i])
+															.toString());
+											if (!selector.isProperty()) {
+												State newState = selector
+														.getState(vals[i]);
+												if (newState != null
+														&& !vals[i].equals("")) {
+													updateState(
+															new ChannelUID(
+																	getThing()
+																			.getUID(),
+																	selector.getChannelID()),
+															newState);
+												} else {
+													updateState(
+															new ChannelUID(
+																	getThing()
+																			.getUID(),
+																	selector.getChannelID()),
+															UnDefType.UNDEF);
+
+												}
+											} else {
+												Map<String, String> properties = editProperties();
+												properties
+														.put(selector
+																.getChannelID(),
+																(selector
+																		.getState(vals[i]))
+																		.toString());
```

Comment:
what exactly do you write into the properties here? Remember that properties are supposed to be static and thus only be set when creating the thing or accessing it for the very first time.
Although no notifications will be triggered, the ThingManager potentially has to persist the properties, so updating them frequently can have very negative impacts.